File: src/test/java/im/tox/tox4j/ConnectedListener.java
package im.tox.tox4j;

import im.tox.tox4j.annotations.NotNull;
import im.tox.tox4j.core.callbacks.ConnectionStatusCallback;
import im.tox.tox4j.core.enums.ToxConnection;

public class ConnectedListener implements ConnectionStatusCallback {
    private @NotNull ToxConnection value = ToxConnection.NONE;

    @Override
    public void connectionStatus(@NotNull ToxConnection connectionStatus) {
        value = connectionStatus;
    }

    public boolean isConnected() {
        return value != ToxConnection.NONE;
    }
}

File: src/test/java/im/tox/tox4j/ConnectedListener.scala
package im.tox.tox4j

import im.tox.tox4j.annotations.NotNull
import im.tox.tox4j.core.callbacks.SelfConnectionStatusCallback
import im.tox.tox4j.core.enums.ToxConnection

final class ConnectedListener extends SelfConnectionStatusCallback {

  @NotNull private var value = ToxConnection.NONE

  override def selfConnectionStatus(@NotNull connectionStatus: ToxConnection): Unit = {
    value = connectionStatus
  }

  def isConnected: Boolean = value != ToxConnection.NONE

}
